### DISCUSSION TITLE
Form related helpers

### DISCUSSION THREAD

**Discussion by KevinBatdorf** [alpine-collective/alpine-magic-helpers#33] [Ideas]
I'm not sure what actually would be useful, but there seems to be a demand for something to help users pre fill forms. 

https://github.com/alpinejs/alpine/issues/872

Personally, I can see when you have a complex form and the server handles the logic of pre filling some fields for you. It might not always be convenient to organize and pass in the fields to the Alpine component to initialize. 

One option is something named `$formFieldModelFinder()`™  that will search for every `x-model` within a component, check if they have a value, see if the value matches the model, then override it or not depending on an option passed in (maybe it only gets overridden if falsey, for example). Any thoughts?

---

I don't often worth with forms. Any other ideas what might be useful? Maybe something captcha related? Or pagination? Errors?

**Comment by SimoTod** [comment]
That's probably the easier way although it doesn't feel Alpine-ish (not declarative, we are setting values for x-models in a different point of the page).
It would be interesting if we could have x-model="$defaultFromValue(foo)" but it won't work because of the way Alpine sets listeners for an x-model directive and because of Alpine not knowing the element which triggered the directive. :(
